Tests/Form/FormFactoryTest.php

```diff
@@ -159,6 +160,47 @@ public function testIsCreatingFormWithTransformers()
         $this->assertEquals('foo_form', $actual);
     }
 
+    public function testIsCreatingFormWithHelpMessages()
+    {
+        $formConfiguration = [
+            'foo' => [
+                'field1' => [
+                    'enabled' => true,
+                    'type' => 'field1_type',
+                    'help_message_provider' => 'cache',
+                    'options' => [
+                        'help' => 'cms:message',
+                    ],
+                ],
+            ],
+        ];
+
+        $helpMessageProviderMock = $this->prophesize('Linio\DynamicFormBundle\HelpMessageProvider');
+
+        $helpMessageProviderMock->getHelpMessage('cms:message')
```

Comment:
Since you don't have `shouldBeCalled()` on most of these, you aren't actually testing that the functionality works.